source/DataNode.cpp

```diff
@@ -271,6 +272,21 @@ bool DataNode::IsBool(const string &token)
 
 
 
+bool DataNode::IsConditionName(const string &token)
```

Comment:
Is there any reason for this to be in DataNode instead of in one of the Condition classes? Data nodes shouldn't even know what conditions are.

Reply:
Yes, I had a number of reasons:
- DataNode already has IsNumber() and IsBool() functions, it seemed to fit as a next item.
- Conditions are used at many locations, I would need to check if there is some sort of central condition class.
- Some non-conditions (like mission names) could also be used as conditions through auto-conditions. We might want to validate their names as well.

But you are right, if this only checks ConditionNames, and not generic names, then it might be better of in a Conditions-specific class.